src/main/Security/EncryptionController.java

```diff
@@ -0,0 +1,151 @@
+package Security;
+
+import Logger.LogFactory;
+import com.google.crypto.tink.Aead;
+import com.google.crypto.tink.JsonKeysetReader;
+import com.google.crypto.tink.KeysetHandle;
+import com.google.crypto.tink.config.TinkConfig;
+import com.google.crypto.tink.integration.gcpkms.GcpKmsClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import org.apache.pdfbox.io.IOUtils;
+import org.bson.Document;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.util.Objects;
+
+import static com.mongodb.client.model.Filters.eq;
+
+public class EncryptionController {
+  Logger logger;
+  MongoDatabase db;
+
+  public static final String masterKeyUri = Objects.requireNonNull(System.getenv("MASTERKEYURI"));
+
+  public static final String credentials =
+      Objects.requireNonNull(System.getenv("GOOGLE_APPLICATION_CREDENTIALS"));
+
+  public EncryptionController(MongoDatabase db) {
+    this.db = db;
+    LogFactory l = new LogFactory();
+    logger = l.createLogger("EncryptionController");
+  }
+
+  // Generates an AEAD Object through Google Tink for encryption and decryption
+  public Aead generateAead() throws GeneralSecurityException, IOException {
+    TinkConfig.register();
+    logger.info("Generating Aead");
+
+    logger.info("Generating Credentials");
+    GoogleCredentials.generateCredentials();
+
+    MongoCollection<Document> keyHandles = db.getCollection("keys", Document.class);
+    Document keyDocument = keyHandles.find(eq("keyType", "encryption")).first();
+
+    assert keyDocument != null;
+    keyDocument.remove("_id");
+    keyDocument.remove("keyType");
+
+    JSONObject keyJson = new JSONObject(keyDocument);
+
+    KeysetHandle keysetHandle =
+        KeysetHandle.read(
+            JsonKeysetReader.withJsonObject(keyJson),
+            new GcpKmsClient().withCredentials(credentials).getAead(masterKeyUri));
+
+    logger.info("KeysetHandle Successfully Generated");
```

Comment:
why do we generate and delete the json credentials on each call?

Reply:
This is a good topic, basically we need to create a temp file keepid-google-kms.json that can be read into this function -> we don't know how to do it without creating a temp file in tink. It is not good practice and I would love to get rid of it. If you have any suggestions please let me know

Reply:
why does the keep.id-google-kms.json need to be temp? why dont we just keep a copy on the heroku server? why even delete it? If we need to rotate it in the future, it would be good to write a query that can delete the old json and generate a new one

Reply:
So, I need to coordinate with you, because actually you are not supposed to use this code on Heroku Google Cloud (we need to switch soon to make this connect), but rather it will automatically connect. This is just so that we can test the code. We are just using for testing.